src/registry/artifactory.rs

```diff
@@ -24,24 +23,50 @@ use semver::Version;
 use serde::Deserialize;
 use url::Url;
 
+/// The policy for validating artifactory server certificates
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum CertValidationPolicy {
+    /// Validate the certificate
+    #[default]
+    Validate,
+
+    /// Do not validate the certificate
+    NoValidation,
+}
+
 /// The registry implementation for artifactory
 #[derive(Debug, Clone)]
 pub struct Artifactory {
-    registry: RegistryUri,
+    registry: url::Url,
```

Comment:
Please make a newtype for holding only valid Artifactory URIs

Reply:
52fe4ef5f2fddf2cba769cfe296609f7ec27800a: Actually restored `RegistryUri` and introduced `RegistryRef`